File: gcj_2014/QR_A.py
#!/usr/bin/env python

# Google Code Jam 2014 Qualification Round
# Problem A. Magic Trick

import sys

def solve(a, b):
  m = []
  for n in a[a[0]]:
    if n in b[b[0]]:
      m.append(n)
  if len(m) == 0:
    return "Volunteer cheated!"
  if len(m) > 1:
    return "Bad magician!"
  return str(m[0])

n = int(sys.stdin.readline())
for t in range(1, n+1):
  args = []
  for n in range(2):
    line = sys.stdin.readline().rstrip("\n")
    arg = [int(line)]
    for i in range(4):
      line = sys.stdin.readline().rstrip("\n")
      arg.append([int(n) for n in line.split(" ")])
    args.append(arg)
#  print(args)
  ans = solve(args[0], args[1])
  print("Case #" + str(t) + ": " + ans)

File: gcj_2022/QR_A.cpp
// Google Code Jam 2022 Qualification Round
// Problem A. Punched Cards (11pts)

#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

vector<string> solve(int r, int c) {
	vector<string> ans(2 * r + 1, string(c * 2 + 1, '+'));
	for (int i = 0; i <= r; ++i) {
		for (int j = 0; j <= c; ++j) {
			if (j < c) {
				ans[i * 2][j * 2 + 1] = '-';
			}
			if (i < r) {
				ans[i * 2 + 1][j * 2] = '|';
			}
			if (i < r && j < c) {
				ans[i * 2 + 1][j * 2 + 1] = '.';
			}
		}
	}
	ans[0][0] = '.';
	ans[0][1] = '.';
	ans[1][0] = '.';
	return ans;
}

int main(int argc, char *argv[]) {
	int T;
	cin >> T;
	for (int t = 1; t <= T; ++t) {
		int r, c;
		cin >> r >> c;
		vector<string> ans = solve(r, c);
		cout << "Case #" << t << ":" << endl;
		for (auto x : ans) {
			cout << x << endl;
		}
	}
	return 0;
}
